airbyte-tests/src/acceptanceTests/java/io/airbyte/test/acceptance/BasicAcceptanceTests.java

```diff
@@ -857,6 +867,168 @@ public void testSyncAfterUpgradeToPerStreamStateWithNoNewData(final TestInfo tes
+      return null;
+    });
+    final ConnectionState initSyncState =
+        apiClient.getConnectionApi().getState(new ConnectionIdRequestBody().connectionId(connection.getConnectionId()));
+    LOGGER.info("ConnectionState after the initial sync: " + initSyncState.toString());
+
+    testHarness.assertSourceAndDestinationDbInSync(false);
+
+    // Patch some data in the source
+    LOGGER.info("Modifying source tables");
+    sourceDb.query(ctx -> {
+      // Adding a new rows to make sure we sync more data.
+      ctx.insertInto(DSL.table(sourceTable1)).columns(DSL.field("name")).values("alice").execute();
+      ctx.insertInto(DSL.table(sourceTable2)).columns(DSL.field("value")).values("v3").execute();
+
+      // The removed rows should no longer be in the destination since we expect a full reset
+      ctx.deleteFrom(DSL.table(sourceTable1)).where(DSL.field("name").eq("john")).execute();
+      ctx.deleteFrom(DSL.table(sourceTable2)).where(DSL.field("value").eq("v2")).execute();
+
+      // Adding a new table to trigger reset from the update connection API
+      ctx.createTableIfNotExists(sourceTable3).columns(DSL.field("location", SQLDataType.VARCHAR)).execute();
+      ctx.truncate(sourceTable3).execute();
+      ctx.insertInto(DSL.table(sourceTable3)).columns(DSL.field("location")).values("home").execute();
+      ctx.insertInto(DSL.table(sourceTable3)).columns(DSL.field("location")).values("work").execute();
+      ctx.insertInto(DSL.table(sourceTable3)).columns(DSL.field("location")).values("space").execute();
+      return null;
+    });
+
+    final AirbyteCatalog updatedCatalog = testHarness.discoverSourceSchemaWithoutCache(sourceId);
+    LOGGER.info("Discovered updated catalog: {}", updatedCatalog);
+
+    // Update with refreshed catalog
+    LOGGER.info("Submit the update request");
+    final WebBackendConnectionUpdate update = new WebBackendConnectionUpdate()
+        .name(connection.getName())
+        .connectionId(connection.getConnectionId())
+        .namespaceDefinition(connection.getNamespaceDefinition())
+        .namespaceFormat(connection.getNamespaceFormat())
+        .prefix(connection.getPrefix())
+        .operations(List.of(
+            new WebBackendOperationCreateOrUpdate()
+                .name(operation.getName())
+                .operationId(operation.getOperationId())
+                .workspaceId(operation.getWorkspaceId())
+                .operatorConfiguration(operation.getOperatorConfiguration())))
+        .syncCatalog(updatedCatalog)
+        .schedule(connection.getSchedule())
+        .sourceCatalogId(connection.getSourceCatalogId())
+        .status(connection.getStatus())
+        .resourceRequirements(connection.getResourceRequirements())
+        .withRefreshedCatalog(true);
+    final WebBackendConnectionRead connectionUpdateRead = webBackendApi.webBackendUpdateConnection(update);
+
+    LOGGER.info("Inspecting Destination DB after the update request, tables should be empty");
```

Comment:
This says that we expect the dest tables to be empty, but we don't actually programmatically enforce that here. Should we add some assertion that the tables are actually empty to codify that?